src/main/resources/api.yaml

```diff
@@ -0,0 +1,308 @@
+      operationId: createNote
+      tags:
+        - notes
+      requestBody:
+        description: Note to create
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Note'
+      responses:
+        '201':
+          description: Note created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Note'
+  /notes/{noteId}:
+    get:
+      summary: Info for a specific note
+      operationId: showNoteById
+      tags:
+        - notes
+      parameters:
+        - name: noteId
+          in: path
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '200':
+          description: A single note
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Note'
+  /quizzes:
+    get:
+      summary: List all quizzes
+      operationId: listQuizzes
+      tags:
+        - quizzes
+      parameters:
+        - name: limit
+          in: query
+          required: false
+          schema:
+            type: integer
+            format: int32
+      responses:
+        '200':
+          description: A list of quizzes
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Quiz'
+    post:
+      summary: Create a quiz
+      operationId: createQuiz
+      tags:
+        - quizzes
+      requestBody:
+        description: Quiz to create
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Quiz'
+      responses:
+        '201':
+          description: Quiz created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Quiz'
+  /quizzes/{quizId}:
+    get:
+      summary: Info for a specific quiz
+      operationId: showQuizById
+      tags:
+        - quizzes
+      parameters:
+        - name: quizId
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: A single quiz
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Quiz'
+  /quizzes/details/{quizId}:
+    get:
+      summary: Details of a specific quiz
+      operationId: showDetailsQuizById
+      tags:
+        - quizzes
+      parameters:
+        - name: quizId
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Details of a single Quiz
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/QuizDetails'
+  /quizzes/details:
```

Comment:
It should have more sense for me if we provide here a quizId - we have an endpoint to create base quiz

Reply:
you mean /quizzes/{quizId}/details?